pkgs/build-support/fetchurl/default.nix

```diff
@@ -1,4 +1,4 @@
-{ stdenv, curl }: # Note that `curl' may be `null', in case of the native stdenv.
+{ stdenv, curl, ca_cert_file ? null, insecure ? false }: # Note that `curl' may be `null', in case of the native stdenv.
```

Comment:
Binary cache handling is done by nix itself, not nixpkgs. There, if all fetches (from one cache) are done in a single connection, the overhead should be pretty minimal. It might be considered to only fetch *.narinfo files securely (tiny), but the performance difference doesn't seem worth the bother (encoding long streams by symmetric ciphers).